shell/platform/windows/flutter_windows_engine.h

```diff
@@ -254,6 +255,19 @@ class FlutterWindowsEngine {
   // Updates accessibility, e.g. switch to high contrast mode
   void UpdateAccessibilityFeatures(FlutterAccessibilityFeature flags);
 
+  // Called when the application quits in response to a quit request.
+  void OnQuit(std::optional<HWND> hwnd,
+              std::optional<WPARAM> wparam,
+              std::optional<LPARAM> lparam,
+              UINT exit_code);
+
+  // Called when a WM_CLOSE message is received.
+  void RequestApplicationQuit(std::optional<HWND> hwnd,
+                              std::optional<WPARAM> wparam,
+                              std::optional<LPARAM> lparam,
```

Comment:
This API is only called when we receive a `WM_CLOSE` right? If so, we will always have `hwnd`, `wparam`, and `lparam` values and these don't need to be `std::optional`. Please let me know if I'm missing something!

Reply:
This is also called when `exitApplication` is called on the app side

Reply:
Gotcha. Please update the comment as it currently indicates it's only for `WM_CLOSE`